src/sentry/similarity.py

```diff
@@ -350,7 +350,7 @@ def serialize_text_shingle(value, separator=b''):
         8,
         2,
         60 * 60 * 24 * 30,
-        4,
+        3,
```

Comment:
This is what the interval structure looks like now for this definition:

```
now	Mon Mar  6 16:21:59 2017
old	Tue Dec  6 16:21:59 2016
1	index	571	lower	Thu Nov 24 16:00:00 2016	upper	Sat Dec 24 16:00:00 2016	expiry	Fri Mar 24 17:00:00 2017
2	index	572	lower	Sat Dec 24 16:00:00 2016	upper	Mon Jan 23 16:00:00 2017	expiry	Sun Apr 23 17:00:00 2017
3	index	573	lower	Mon Jan 23 16:00:00 2017	upper	Wed Feb 22 16:00:00 2017	expiry	Tue May 23 17:00:00 2017
4	index	574	lower	Wed Feb 22 16:00:00 2017	upper	Fri Mar 24 17:00:00 2017	expiry	Thu Jun 22 17:00:00 2017
```

You can check the math by ensuring a retention of 0 results a single bucket with an expiration that is equal to the upper bound (why anyone would do this, I don't know):

```
new	Mon Mar  6 16:26:13 2017
old	Mon Mar  6 16:26:13 2017
1	index	574	lower	Wed Feb 22 16:00:00 2017	upper	Fri Mar 24 17:00:00 2017	expiry	Fri Mar 24 17:00:00 2017
```